src/extension/CMakeLists.txt

```diff
@@ -14,45 +14,192 @@ add_library(thrive_extension SHARED
   core/GodotJoltConversions.hpp
 )
 
-if(MSVC)
-  target_compile_options(thrive_extension PRIVATE /W4 /wd4068)
+target_sources(thrive_extension PRIVATE ${EXTENSION_SOURCES})
 
-  if(WARNINGS_AS_ERRORS)
-    target_compile_options(thrive_extension PRIVATE /WX)
-  endif()
+# Print files and configure build
+print_source_files(thrive_extension)
```

Comment:
I really think it should be a special `option` to print the source files and not do it by default.